x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeKeywordMappedFieldType.java

```diff
@@ -77,8 +78,17 @@ public String typeName() {
         return scriptFactory.newFactory(script.getParams(), context.lookup());
     }
 
+    private void checkAllowExpensiveQueries(QueryShardContext context) {
+        if (context.allowExpensiveQueries() == false) {
+            throw new IllegalArgumentException(
+                "queries cannot be executed against [script] fields while [" + ALLOW_EXPENSIVE_QUERIES.getKey() + "] is set to [false]."
```

Comment:
nit: use the constant from the mapper? content type I think it is called

Reply:
should we rather throw ElasticsearchException also, and why doesn't the context expose a check method instead that accepts for instance a message? :)

Reply:
> should we rather throw ElasticsearchException also, and why doesn't the context expose a check method instead that accepts for instance a message? :)

That would probably be a good choice.

I have no idea why we were throwing ElasticsearchException when we were. It feels like this is what IAE is for.

Reply:
> I have no idea why we were throwing ElasticsearchException when we were. It feels like this is what IAE is for.

I figured it out - we translate all exceptions thrown when building a query into a `QueryShardException` which is always a 400.

Reply:
> why doesn't the context expose a check method instead that accepts for instance a message? :)

It really is a good idea. I took a look at it and, ironically, testing it is kind of a pain. I'll hold off on it.

Reply:
shall we open issues for possible follow-up tasks that we don't want to spend time on right now?

Reply:
Let me give it another go this morning!